alacritty/src/input/keyboard.rs

```diff
@@ -172,7 +172,13 @@ impl<T: EventListener, A: ActionContext<T>> Processor<T, A> {
             // the time. However what we want is to manually lowercase the character to account
             // for both small and capital letters on regular characters at the same time.
             let logical_key = if let Key::Character(ch) = key.logical_key.as_ref() {
-                Key::Character(ch.to_lowercase().into())
+                // Match `Alt` bindings without `Alt` being applied, otherwise they use the
+                // composed chars, which are not intuitive to bind.
+                if cfg!(target_os = "macOS") && mods.alt_key() {
+                    key.key_without_modifiers()
```

Comment:
This will also remove other modifiers though, right? That just seems extremely inconsistent and confusing.

Reply:
That's what we're doing though already, we just try to not remove `Shift`, but `Alt` breaks `Shift` on macOS anyway, thus it doesn't really matter.

We're not using `text_with_all_modifiers` or anything.

Reply:
That's how it works before if it matters when it comes to logical keys on macOS and that's probably the reason we have `Alt + digit` not working and similar.

Reply:
> but Alt breaks Shift on macOS anyway, thus it doesn't really matter.

But alt only breaks shift if it's configured to do so, right? If alt_send_esc is enabled for both it will not?

Reply:
It'll not break if it's just shift and `alt_send_esc`, when you have control or super, it'll still break. regardless.

I'm not sure there's a clear way to solve this and I don't see much issues in the proposed approach for macOS, since obviously you also want your bindings to work when you don't have `option_as_alt`...